staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2313,6 +2314,33 @@ type SeccompConfiguration struct {
 	VirtualMachineInstanceProfile *VirtualMachineInstanceProfile `json:"virtualMachineInstanceProfile,omitempty"`
 }
 
+// MemoryOverCommitmentProfile holds memory over commitment profile for Kubevirt.
+type MemoryOverCommitmentProfile struct {
+	// +kubebuilder:validation:Enum=Off;Baseline;Insecure
+	Strategy *Strategy             `json:"strategy,omitempty"`
+	Custom   *MemoryOverCommitment `json:"custom,omitempty"`
+}
+
+// MemoryOverCommitment holds memory over commitment fields
+type MemoryOverCommitment struct {
+	// FreePageReporting enable or disable the free page reporting of
+	// memory balloon device https://libvirt.org/formatdomain.html#memory-balloon-device.
+	// This will have effect only if AutoattachMemBalloon is not false.
+	// Default: false
+	FreePageReporting *bool `json:"freePageReporting,omitempty"`
+}
+
+type Strategy string
+
+const (
+	// Off with freePageReporting enabled.
+	Off Strategy = "Off"
+	// Baseline is a good baseline with freePageReporting enabled.
+	Baseline Strategy = "Baseline"
+	// Insecure should only be used with trusted workloads.
+	Insecure Strategy = "Insecure"
```

Comment:
TBH I don't really understand how these are different and what they stand for.
Maybe we can add only `Baseline` if all of them do the same thing as for now?

Or even better - how about allowing an empty `MemoryOverCommitmentProfile` to indicate that user doesn't care about the exact strategy and wants a sane default?

Reply:
> > I added a reference in the field description. I'd prefer not to add a constraint between `freePageReporting` and `autoattachMemBallon`, since the freePageReporting value will be ignored if autoattachMemBalloon is false. From this point of view I cannot see any drawback.
> 
> My concern is that we would take two different paths when freePageReporting is being defined at vmi level and the Kubevirt CR level. IOW, what would happen if freePageReporting is being set on the CR level but a VMI doesn't have a virtio-balloon driver? Obviously it doesn't make sense to fail the vmi creation because of that.
> 
> If it's being silently ignored, perhaps a vmi-level warning would be enough?

@iholder101 That's a good point! Do you mean a condition or an event?

Reply:
Thank you!

> Strategies will differ each other from my point of view with future enhancements :)

Waiting for future enhancements :)

In general, I think that for most APIs it makes sense to not specify anything for the user to say "I don't care about the strategy" so we would know the user didn't explicitly choose a strategy, even if it's the default